### ISSUE TITLE
more "couldn't determine unixfs type" samples

### ISSUE THREAD

**Issue by Arqu** [n0-computer/iroh#351]
/ipfs/bafyreifarfukpxryrrk37aye2pktmwelt4yq7ddvzkg75rbytiwjxhpyne/metadata.json
/ipfs/bafyreicg6ztxll62hg3vfe7firant5eydbpwfmr6uffxmeky46o5bc2vgu/metadata.json

noticed on the `bitswap2` dunno if it still applies on main.

**Comment by dignifiedquire** [comment]
`ipfs/bafybeiemxf5abjwjbikoz4mc3a3dla6ual3jsgpdr4cjr3oz3evfyavhwq/wiki/` fails with 

```json
{"code":500,"message":"failed to read dir listing","success":false,"trace_id":"00000000000000000000000000000000"}
```

**Comment by Arqu** [comment]
a bunch more like this
```Oct 17 14:10:29 ip-172-31-82-125 teststream[226969]: REZ: 500 | g-dig | {"code":500,"message":"unexpected unixfs type Some(File)","success":false,"trace_id":"00000000000000000000000000000000"} | http://g1.gateway.lol:9050/ipfs/bafybeigy6zyjllz3lsbu5exsrgks54utgsvuvddphaytrbmypevya7lkuy%2F```

**Comment by ramfox** [comment]
> a bunch more like this `Oct 17 14:10:29 ip-172-31-82-125 teststream[226969]: REZ: 500 | g-dig | {"code":500,"message":"unexpected unixfs type Some(File)","success":false,"trace_id":"00000000000000000000000000000000"} | http://g1.gateway.lol:9050/ipfs/bafybeigy6zyjllz3lsbu5exsrgks54utgsvuvddphaytrbmypevya7lkuy%2F`

fwiw `main` resolves `/ipfs/bafybeigy6zyjllz3lsbu5exsrgks54utgsvuvddphaytrbmypevya7lkuy` just fine on the local gateway. Will investigate this again when we merge the latest bitswap stuff



> `ipfs/bafybeiemxf5abjwjbikoz4mc3a3dla6ual3jsgpdr4cjr3oz3evfyavhwq/wiki/` fails with
> 
> ```json
> {"code":500,"message":"failed to read dir listing","success":false,"trace_id":"00000000000000000000000000000000"}
> ```

This one I'm less inclined to investigate: the target is ~500 GiB

Primarily looking into the errors for these:
`/ipfs/bafyreicg6ztxll62hg3vfe7firant5eydbpwfmr6uffxmeky46o5bc2vgu/metadata.json`

I'm getting the usual `couldn't determine unixfs type` errors on the gateway, and some very odd errors when trying to get using the command line. Basically, we seem to be expecting one block of `UnixfsNode::File` type, but receive multiple blocks & in an unexpected order.

**Comment by ramfox** [comment]
`/ipfs/bafybeigy6zyjllz3lsbu5exsrgks54utgsvuvddphaytrbmypevya7lkuy` gives a different filesize when downloaded via the command line and the gateway. The gateway filesize matches what is downloaded from the ipfs.io gateway & `ipfs get` command.

**Comment by dignifiedquire** [comment]
getting

```json
{"code":500,"message":"UnixfsNode::HamtShard link '' not found","success":false}
```

on 
- `http://localhost:9050/ipfs/bafybeiemxf5abjwjbikoz4mc3a3dla6ual3jsgpdr4cjr3oz3evfyavhwq/-/j/`
but 
- `http://localhost:9050/ipfs/bafybeiemxf5abjwjbikoz4mc3a3dla6ual3jsgpdr4cjr3oz3evfyavhwq/-/j`

works fine

**Comment by Arqu** [comment]
```
Oct 20 13:47:05 ip-172-31-82-125 teststream[249059]: REZ: 500 | g-dig | {"code":500,"message":"Expected Key(\"\") but found Bytes","success":false} | http://g1.gateway.lol:9050/ipfs/bafkreiepsojuhcfji7wwdy2npx7egyhi66aknogiklq35hiurynedw3fuu%2F
Oct 20 13:47:05 ip-172-31-82-125 teststream[249059]: REZ: 500 | g-dig | {"code":500,"message":"Expected Key(\"\") but found Bytes","success":false} | http://g1.gateway.lol:9050/ipfs/bafkreihr656dzh7sv75ntn3voehvuo3phucrdvezq2yzhyu5tiu7nlbs64%2F
Oct 20 13:47:05 ip-172-31-82-125 teststream[249059]: REZ: 500 | g-dig | {"code":500,"message":"Expected Key(\"\") but found Bytes","success":false} | http://g1.gateway.lol:9050/ipfs/bafkreihr656dzh7sv75ntn3voehvuo3phucrdvezq2yzhyu5tiu7nlbs64%2F
Oct 20 13:47:05 ip-172-31-82-125 teststream[249059]: REZ: 500 | g-dig | {"code":500,"message":"Expected Key(\"\") but found Bytes","success":false} | http://g1.gateway.lol:9050/ipfs/bafkreifuerb2m345xndnrncg4zw5k7pcukxlixnvayuc2h7kqil4r2rgwa%2F
Oct 20 13:47:06 ip-172-31-82-125 teststream[249059]: REZ: 500 | g-dig | {"code":500,"message":"unexpected unixfs type Some(File)","success":false} | http://g1.gateway.lol:9050/ipfs/bafybeiagt5yoomhejtuuv4kulkc3sqf6yxsky7ai7s6fvbktb4v34bat6y%2F
Oct 20 13:47:06 ip-172-31-82-125 teststream[249059]: REZ: 500 | g-dig | {"code":500,"message":"unexpected unixfs type Some(File)","success":false} | http://g1.gateway.lol:9050/ipfs/bafybeihwzkjkzlj2pb42buq2pptyqqepm6oe2miebsd3ncmyya7qqk2kbq%2F
Oct 20 13:47:07 ip-172-31-82-125 teststream[249059]: REZ: 500 | g-dig | {"code":500,"message":"unexpected unixfs type Some(File)","success":false} | http://g1.gateway.lol:9050/ipfs/bafybeidjmdkgc3shpwpr2if4gi3hydlbh67lxj5swtdozd3llvj4demmie%2F
Oct 20 13:47:08 ip-172-31-82-125 teststream[249059]: REZ: 500 | g-dig | {"code":500,"message":"unexpected unixfs type Some(File)","success":false} | http://g1.gateway.lol:9050/ipfs/bafybeigvahkaljqn46cio44sgknytx7cccupykp4xxdlknvl64kyeu2fia%2F
Oct 20 13:47:08 ip-172-31-82-125 teststream[249059]: REZ: 500 | g-dig | {"code":500,"message":"unexpected unixfs type Some(File)","success":false} | http://g1.gateway.lol:9050/ipfs/bafybeicgbrk5i3bubo5ggqg7uww7sh5ohcu2lyzdcpiv2h5yvh76vfsa6i%2F
Oct 20 13:47:08 ip-172-31-82-125 teststream[249059]: REZ: 500 | g-dig | {"code":500,"message":"unexpected unixfs type Some(File)","success":false} | http://g1.gateway.lol:9050/ipfs/bafybeigupqomatfo4kxceye62p5taflnzrvvsatna4egivvf2fnlitf5vq%2F
Oct 20 13:47:08 ip-172-31-82-125 teststream[249059]: REZ: 500 | g-dig | {"code":500,"message":"unexpected unixfs type Some(File)","success":false} | http://g1.gateway.lol:9050/ipfs/QmT1QMTy7qN4KanNLFC86dRLwZSnhhCEBwjwuCVsYUiJcq%2F
Oct 20 13:47:08 ip-172-31-82-125 teststream[249059]: REZ: 500 | g-dig | {"code":500,"message":"unexpected unixfs type Some(File)","success":false} | http://g1.gateway.lol:9050/ipfs/bafybeih7bqlltkyhdkvf35xwfolwcqdkhbl3yihek7wf6e3nul3pa4gtma%2F
Oct 20 13:47:09 ip-172-31-82-125 teststream[249059]: REZ: 500 | g-dig | {"code":500,"message":"unexpected unixfs type Some(File)","success":false} | http://g1.gateway.lol:9050/ipfs/bafybeihrm7xgno2h77wqh4g7qsqd7gpeg66a3jnnrhdaosc3vmdebbvlke%2F
Oct 20 13:47:10 ip-172-31-82-125 teststream[249059]: REZ: 500 | g-dig | {"code":500,"message":"unexpected unixfs type Some(File)","success":false} | http://g1.gateway.lol:9050/ipfs/bafybeibama3bfeh5g2ne5eedwmjqyhnxszwudma3f7dsgffa2tnlhiiwpe%2F
Oct 20 13:47:10 ip-172-31-82-125 teststream[249059]: REZ: 500 | g-dig | {"code":500,"message":"unexpected unixfs type Some(File)","success":false} | http://g1.gateway.lol:9050/ipfs/bafybeih2ay6or6hbm4wdyogjhh3ftimkszqkvsm5kcyufshhw6wetumevq%2F
```

**Comment by b5** [comment]
is this fixed by #408?

**Comment by ramfox** [comment]
yes, not sure why it didn't get picked up by the pr close. Closing now!
